### DISCUSSION TITLE
why is laravel asking me for the role_user table

### DISCUSSION THREAD

**Discussion by edat86** [spatie/laravel-permission#2009] [Q&A]
Error I have is when I want to assign the role to a user.
assignRole()

 • A table was not found: You might have forgotten to run your migrations. You can run your migrations using php artisan migrate.
    https://laravel.com/docs/master/migrations#running-migrations

  1   ......................\vendor\laravel\framework\src\Illuminate\Database\Connection.php:368
      PDOException::("SQLSTATE[42S02]: Base table or view not found: 1146 Table 'role_user' doesn't exist")

  2   ......................\vendor\laravel\framework\src\Illuminate\Database\Connection.php:368
      PDO::prepare("select * from role_user where role_user.user_id = ?")

**Comment by erikn69** [comment]
It is your implementation, this package does not has role_user table
[must-not-have-a-role-or-roles-property-nor-a-roles-method](https://spatie.be/docs/laravel-permission/v5/prerequisites#must-not-have-a-role-or-roles-property-nor-a-roles-method)
